### PR TITLE
feat: connect/disconnect to/from google

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: #2211

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

Go to https://deploy-preview-2221--regen-marketplace.netlify.app/ and log in.
Go to the settings page and try connecting/disconnect to/from google.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
@erikalogie see testing instructions above

**Comment by erikalogie** [comment]
@blushi I'm having trouble connecting with google

https://github.com/regen-network/regen-web/assets/8460237/affc0cab-882d-401f-bb9c-9451322e9979

**Comment by aaronc** [comment]
I tried connecting to google and just get this:
image

**Comment by blushi** [comment]
> I tried connecting to google and just get this: image

It looks as if we were not logged in (only logged in users can connect an account to google, hence the unauthorized error: https://github.com/regen-network/regen-server/blob/fd79a95da729e64ef2e98d5340d89abc80fe376a/server/routes/auth.ts#L41). Could you describe how you logged in before trying to connect to google?

**Comment by blushi** [comment]
> @blushi I'm getting this when I first log in with my email and navigate to the settings. These two things should be congruent (there should be a connect button not a disconnect button) https://monosnap.com/file/uvg5HJuextai7Wec0lSmHjI5uvTH9j

Looks like this might be a one-time issue based on the recent database migration for adding a specific google_email that can be different from the main email. This should not happen again.

**Comment by blushi** [comment]
> > I tried connecting to google and just get this: image
> 
> It looks as if you were not logged in (only logged in users can connect an account to google, hence the unauthorized error: https://github.com/regen-network/regen-server/blob/fd79a95da729e64ef2e98d5340d89abc80fe376a/server/routes/auth.ts#L41). Could you describe how you logged in before trying to connect to google? which browser are you using?

Nevermind, I could reproduce it using Brave so I'll investigate.

**Comment by blushi** [comment]
@erikalogie @aaronc it looks like this is because Brave blocks inter-domain cookies by default (which is the case for deploy previews which have a different domain than the server, it should work fine on https://dev.app.regen.network/). In order to test google connect on deploy previews, you will need to go to the Brave Shields settings and authorize all cookies for the deploy preview.

**Comment by erikalogie** [comment]
> @erikalogie @aaronc it looks like this is because Brave blocks inter-domain cookies by default (which is the case for deploy previews which have a different domain than the server, it should work fine on https://dev.app.regen.network/), even if they are set as `Secure` like it's the case here in staging. In order to test google connect on deploy previews, you will need to go to the Brave Shields settings and authorize all cookies for the deploy preview.

This is working for me now!

File: web-marketplace/src/pages/ProfileEdit/ProfileEdit.Settings.tsx [hunk 1]
```diff
@@ -38,10 +38,10 @@ export const ProfileEditSettings = () => {
   } = useLoginData();
 
   // Social providers
+  const socialProviders = useSocialProviders();
   const _socialProviders = socialProviders.map(p => ({
     name: p.name,
-    // TODO: we'll need to replace this with the email from the provider account #2211
-    email: privActiveAccount?.email,
+    email: privActiveAccount?.[`${p.id}_email`],
```
Comment (flagrede) [inline web-marketplace/src/pages/ProfileEdit/ProfileEdit.Settings.tsx:44]: Should we avoid using single-character variables even for temporary ones. Using `socialProvider` here might be more visible.

**Review by flagrede — APPROVED** [review]
Looks good!
One minor comment.

**Pull request merged**